### PR TITLE
Fix Kafka receiver closing ready channel multiple times

### LINKED ISSUE #1692: kafka receiver panic: close of closed channel
**Describe the bug**
I use kafkareceiver to receive data. When running for a period of time (about 10s), the program will panic.

```
panic: close of closed channel

goroutine 201 [running]:
go.opentelemetry.io/collector/receiver/kafkareceiver.(*consumerGroupHandler).Setup(0xc0000ae880, 0x1b8eea0, 0xc000798200, 0xc001d50660, 0xc000798200)
        /Users/niaoge/go/pkg/mod/go.opentelemetry.io/collector@v0.9.1-0.20200831045722-c3f7dd5e40a3/receiver/kafkareceiver/kafka_receiver.go:131 +0x36
github.com/Shopify/sarama.newConsumerGroupSession(0x1b79980, 0xc001544280, 0xc00002a3f0, 0xc001684150, 0xc0000100c0, 0x3b, 0x1c4f, 0x1b74dc0, 0xc0000ae880, 0x0, ...)
        /Users/niaoge/go/pkg/mod/github.com/!shopify/sarama@v1.27.0/consumer_group.go:597 +0x464
github.com/Shopify/sarama.(*consumerGroup).newSession(0xc00002a3f0, 0x1b79980, 0xc0000ae840, 0xc0001d2610, 0x1, 0x1, 0x1b74dc0, 0xc0000ae880, 0x4, 0xc00004a280, ...)
        /Users/niaoge/go/pkg/mod/github.com/!shopify/sarama@v1.27.0/consumer_group.go:299 +0x2f7
github.com/Shopify/sarama.(*consumerGroup).Consume(0xc00002a3f0, 0x1b79980, 0xc0000ae840, 0xc0001d2610, 0x1, 0x1, 0x1b74dc0, 0xc0000ae880, 0x0, 0x0)
        /Users/niaoge/go/pkg/mod/github.com/!shopify/sarama@v1.27.0/consumer_group.go:169 +0x18b
go.opentelemetry.io/collector/receiver/kafkareceiver.(*kafkaConsumer).consumeLoop(0xc0001e0690, 0x1b79980, 0xc0000ae840, 0x1b74dc0, 0xc0000ae880, 0x8c2c25, 0xc000022630)
        /Users/niaoge/go/pkg/mod/go.opentelemetry.io/collector@v0.9.1-0.20200831045722-c3f7dd5e40a3/receiver/kafkareceiver/kafka_receiver.go:103 +0x9b
created by go.opentelemetry.io/collector/receiver/kafkareceiver.(*kafkaConsumer).Start
        /Users/niaoge/go/pkg/mod/go.opentelemetry.io/collector@v0.9.1-0.20200831045722-c3f7dd5e40a3/receiver/kafkareceiver/kafka_receiver.go:93 +0x169
```

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <ploffay@redhat.com>

**Description:** <Describe what has changed. 

Avoid closing ready channel multiple times.

**Link to tracking Issue:** <Issue number if applicable>

Fixes #1692 

**Testing:** < Describe what testing was performed and which tests were added.>

**Documentation:** < Describe the documentation added.>

### CODE REVIEW

File: consumer/pdata/generated_metrics.go [hunk 1]
```diff
@@ -607,8 +607,6 @@ func (ms Metric) SetUnit(v string) {
 	(*ms.orig).Unit = v
 }
 
-
```
Comment (pavolloffay) [inline consumer/pdata/generated_metrics.go:610]: `make fmt` seems to touch this file. I can revert if needed.

Comment (bogdandrutu) [reply consumer/pdata/generated_metrics.go:610]: It shouldn't anymore.

**Comment by bogdandrutu** [comment]
There is a data race in the test

File: receiver/kafkareceiver/kafka_receiver.go [hunk 2]
```diff
@@ -121,14 +122,17 @@ type consumerGroupHandler struct {
...
+	readyCloser sync.Once
 
 	logger *zap.Logger
 }
 
 var _ sarama.ConsumerGroupHandler = (*consumerGroupHandler)(nil)
 
 func (c *consumerGroupHandler) Setup(session sarama.ConsumerGroupSession) error {
-	close(c.ready)
+	c.readyCloser.Do(func() {
```
Comment (pavolloffay) [inline receiver/kafkareceiver/kafka_receiver.go:133]: The setup can be called multiple times 

https://github.com/Shopify/sarama/issues/1793#issuecomment-683900202

> There is an issue with the example, as the Setup method could be called multiple times as part of when a member initially joins a group or a rebalance happens.

The example from sarama https://github.com/Shopify/sarama/blob/master/examples/consumergroup/main.go#L142 reinitializes the ready channel in the consumer loop that leads to data race. For us it's enough to close the ready channel once as it is used only once in receiver start method to block until receiver is ready to consumer mesages.

**Comment by pavolloffay** [comment]
@bogdandrutu this is ready for review.

**Pull request merged**